### PR TITLE
feat: add email blocklist section with mocked data

### PR DESCRIPTION
**Pull request by lakshay-saini-au8**
https://github.com/twentyhq/twenty/issues/2832

1. Added Settings/Accounts/Emails page a "Blocklist" section with a title, a description, an input, an "Add to blocklist" button, and a table that displays blocked emails.
2. Onclick  "Add to blocklist" button add a row in a state local to the component for now.

https://github.com/twentyhq/twenty/assets/19304527/f042c6b3-d502-4bb2-b518-bef11f2c079d

### CODE REVIEW

**Comment by lakshay-saini-au8** [comment]
@thaisguigon @charlesBochet 
Please review

File: packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsEmailsBlocklistSection.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  const handleBlockedEmailRemove = (id: string) =>
+    setBlockedEmailList((previousBlockedEmailList) =>
+      previousBlockedEmailList.filter((blockedEmail) => blockedEmail.id !== id),
+    );
+
+  const updateBlockedEmailList = (email: string) =>
+    setBlockedEmailList((prevState) => [
+      ...prevState,
+      { id: v4(), email: email, blocked_at: '21/12/2023' },
+    ]);
```
Comment (bosiraphael) [inline packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsEmailsBlocklistSection.tsx:24]: Later we will need to check if the value `email` is indeed an email and we will also need to check if the email isn't already blocked.
I think that even if these are mock data, we could put the current date in `blocked_at`

File: packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsEmailsBlocklistInput.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  const [formValues, setFormValues] = useState<{
+    email: string;
+  }>({
+    email: '',
+  });
+  return (
+    <StyledContainer>
+      <StyledLinkContainer>
+        <TextInput
+          placeholder="eddy@gmail.com"
```
Comment (bosiraphael) [inline packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsEmailsBlocklistInput.tsx:34]: According to the Figma, the placeholder should be "eddy@gmail.com, @apple.com".

**Review by bosiraphael — COMMENTED** [review]
@lakshay-saini-au8 Thank you very much for contributing. I left a couple of comments, but great work overall :)

File: packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsEmailsBlocklistInput.tsx (hunk 2 above)
Comment (lakshay-saini-au8) [reply packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsEmailsBlocklistInput.tsx:34]: Done

File: packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsEmailsBlocklistSection.tsx (hunk 1 above)
Comment (lakshay-saini-au8) [reply packages/twenty-front/src/modules/settings/accounts/components/SettingsAccountsEmailsBlocklistSection.tsx:24]: Yeah, we can put the check later on once we integrate with the backend, also I didn't find any error state design for this in Figma.

**Comment by lakshay-saini-au8** [comment]
@bosiraphael Please review

Screenshot 2023-12-27 at 12 53 41 PM

**Review by bosiraphael — APPROVED** [review]
LGTM, thank you !

**Pull request merged**